libparsec/crates/client/src/workspace/store/folder_updater.rs

```diff
@@ -60,33 +62,43 @@ pub(super) async fn for_update_folder(
             listener.await;
         }
 
-        let update_guard = {
-            let mut cache_guard = store.current_view_cache.lock().expect("Mutex is poisoned");
-
-            // 1) Lock for update
-
-            let outcome = cache_guard.lock_update_manifests.take(entry_id);
-            match outcome {
+        enum LockForUpdateOutcome {
```

Comment:
Isn't that the same struct as in `file_updater.rs` ?

Reply:
Yes, but I don't think it is worth it to factorize it since this is purely implementation detail of `for_update_*` code (that's why it is defined within the function body).

Basically factorize just a type would create an artificial relationship between two unrelated pieces of code that use it, which typically cause issue when modifying one piece but not the other (on top of adding complexity when reading the code since the type definition end up in another file).

However you're right on the fact both `for_update_*` functions have very similar implementations, so we should be able to factorize the entire implementation. I will look into this in my next PR (see #9153) that will modify part of this code anyway.